internal/pkg/composable/providers/kubernetes/hints.go

```diff
@@ -256,17 +256,48 @@ func GetHintsMapping(k8sMapping map[string]interface{}, logger *logp.Logger, pre
 		composableMapping: mapstr.M{},
 		processors:        []mapstr.M{},
 	}
+	var hints mapstr.M
+	var containerProcessors []mapstr.M
 
 	if ann, ok := k8sMapping["annotations"]; ok {
 		annotations, _ := ann.(mapstr.M)
-		hints := utils.GenerateHints(annotations, "", prefix)
-		if len(hints) > 0 {
-			logger.Debugf("Extracted hints are :%v", hints)
-			hintData.composableMapping = GenerateHintsMapping(hints, k8sMapping, logger, cID)
-			logger.Debugf("Generated hints mappings are :%v", hintData.composableMapping)
-
-			hintData.processors = utils.GetConfigs(annotations, prefix, processorhints)
-			logger.Debugf("Generated Processors are :%v", hintData.processors)
+
+		if containerEntries, err := annotations.GetValue(prefix + ".hints"); err == nil {
+			entries, _ := containerEntries.(mapstr.M)
+			if len(entries) > 0 {
+				for key := range entries {
+					parts := strings.Split(key, "/")
+					if con, ok := k8sMapping["container"]; ok {
+						containers, _ := con.(mapstr.M)
+						if cname, err := containers.GetValue("name"); err == nil {
+							if parts[0] == cname {
+								// If there are hints like co.elastic.hints.<container_name>/ then add add the values after the / to the corresponding container
+								hints = utils.GenerateHints(annotations, parts[0], prefix)
+								// Processors for specific container
+								// We need to make an extra check if we have processors added only to the specific containers
+								containerProcessors = utils.GetConfigs(annotations, prefix, "hints."+parts[0]+"/processors")
+
+							} else {
+								// If there are top level hints like co.elastic.hints/ then just add the values after the /
```

Comment:
So is that true that we either support only container's hints or top level hints? Couldn't be a case that we want to support both at the same time?

Reply:
We support both at the same time